dot/network/service.go

```diff
@@ -130,6 +133,18 @@ func NewService(cfg *Config) (*Service, error) {
 		return nil, err
 	}
 
+	// pre-allocate pool of buffers used to read from streams.
+	// initially allocate as many buffers as minimally necessary which is the number inbound streams we will have,
+	// which should equal min peers times the number of notifications protocols, which is currently 3.
+	var bufPool *sizedBufferPool
+	if !cfg.noPreAllocate {
```

Comment:
If you switch `if` with `else` it will be cleaner.

if cfg.noPreAllocate {
    bufPool = newSizedBufferPool(cfg.MinPeers*3, cfg.MaxPeers*3)
} else {
    bufPool = &sizedBufferPool{
	c: make(chan *[maxMessageSize]byte, cfg.MaxPeers*3),
    }
}